integrations/opsgenie/client.py

```diff
@@ -0,0 +1,153 @@
+                )
+                raise
+
+    async def get_alert(self, identifier: str) -> dict[str, Any]:
+        api_version = await self.get_resource_api_version(ObjectKind.ALERT)
+        url = f"{self.api_url}/{api_version}/alerts/{identifier}"
+        alert_data = (await self._get_single_resource(url))["data"]
+        return await self.get_related_incident_by_alert(alert_data)
+
+    async def get_oncall_team(self, identifier: str) -> dict[str, Any]:
+        cache_key = f"{ObjectKind.TEAM}-{identifier}"
+        if cache := event.attributes.get(cache_key):
+            return cache
+        api_version = await self.get_resource_api_version(ObjectKind.TEAM)
+        url = f"{self.api_url}/{api_version}/teams/{identifier}"
+        oncall_team = (await self._get_single_resource(url))["data"]
+        event.attributes[cache_key] = oncall_team
+        return oncall_team
+
+    async def get_oncall_user(self, schedule_identifier: str) -> dict[str, Any]:
+        api_version = await self.get_resource_api_version(ObjectKind.SCHEDULE)
+        url = f"{self.api_url}/{api_version}/schedules/{schedule_identifier}/on-calls?flat=true"
+        return (await self._get_single_resource(url))["data"]
+
+    async def get_cached_resources(
+        self, resource_type: ObjectKind
+    ) -> list[dict[str, Any]]:
+        resources_list = []
+        if cache := event.attributes.get(resource_type.value):
+            return cache
+        async for batch_data in self.get_paginated_resources(resource_type):
+            resources_list.extend(batch_data)
+        event.attributes[resource_type.value] = resources_list
+        return resources_list
+
+    async def get_schedule_by_team(
+        self, team_identifier: str
+    ) -> Optional[dict[str, Any]]:
+        schedules = await self.get_cached_resources(ObjectKind.SCHEDULE)
+        for schedule in schedules:
+            if schedule["ownerTeam"]["id"] == team_identifier:
+                return schedule
+        return {}
+
+    async def get_associated_alerts(
+        self, incident_identifier: str
+    ) -> list[dict[str, Any]]:
+        cache_key = f"{ObjectKind.INCIDENT}-{incident_identifier}"
+        if cache := event.attributes.get(cache_key):
+            return cache
+
+        api_version = await self.get_resource_api_version(ObjectKind.INCIDENT)
+        url = f"{self.api_url}/{api_version}/incidents/{incident_identifier}/associated-alert-ids"
+        associated_alerts = (await self._get_single_resource(url))["data"]
+        event.attributes[cache_key] = associated_alerts
+        return associated_alerts
+
+    async def get_incidents_by_service(self, service_id: str) -> list[dict[str, Any]]:
+        incidents = await self.get_cached_resources(ObjectKind.INCIDENT)
+        impacted_services = []
+        for incident in incidents:
+            if service_id in incident["impactedServices"]:
+                impacted_services.append(incident)
+        return impacted_services
```

Comment:
```suggestion
return [incident for incident in incidents if service_id in incident["impactedServices"]]
```

Reply:
I read the [ocean development guideline](https://ocean.getport.io/develop-an-integration/guidelines#performance) and it suggested that developers use simple syntax. That's why I have been avoiding list comprehensions

Reply:
list comprehension as long as you keep it with one for loop
its a more pythonic way to do it